prow/plugins/config.go

```diff
@@ -1789,6 +1789,25 @@ func (c *Configuration) mergeFrom(other *Configuration) error {
 	c.Approve = append(c.Approve, other.Approve...)
 	c.Lgtm = append(c.Lgtm, other.Lgtm...)
 
+	if err := c.mergeExternalPluginsFrom(other.ExternalPlugins); err != nil {
+		errs = append(errs, fmt.Errorf("failed to merge .external-plugins from supplemental config: %w", err))
+	}
+
+	return utilerrors.NewAggregate(errs)
+}
+
+func (c *Configuration) mergeExternalPluginsFrom(other map[string][]ExternalPlugin) error {
+	var errs []error
+	for orgOrRepo, config := range other {
+		if _, ok := c.ExternalPlugins[orgOrRepo]; ok {
+			errs = append(errs, fmt.Errorf("found duplicate config for external-plugins.%s", orgOrRepo))
+			continue
+		}
+		if c.ExternalPlugins != nil {
```

Comment:
We want to always merge the config, not only when the main config already has config. You need to initialize the map when its nil, not gate the writing to the map by a nilcheck. Please also add a testcase for this (main config has no `ExtternalPlugins` config, supplemental config has, it gets merged)